packages/eslint-plugin/src/rules/no-array-delete.ts

```diff
@@ -0,0 +1,57 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import * as ts from 'typescript';
+
+import * as util from '../util';
+
+export default util.createRule<[], 'arrayDeleteViolation'>({
```

Comment:
Nit: no need for explicit type parameters here. They can be inferred:

```suggestion
export default util.createRule({
```